testing/internal/e2e/tests/static/authenticate_primary_test.go

```diff
@@ -0,0 +1,62 @@
+package static_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/creack/pty"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCliAuthenticatePrimary tests the `boundary authenticate` command to ensure that a user can
+// more quickly log in using the short-hand form of the command (authenticates using the primary auth
+// method).
+func TestCliAuthenticatePrimary(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	bc, err := boundary.LoadConfig()
+	require.NoError(t, err)
+
+	var cmd *exec.Cmd
+	ctx := context.Background()
+	cmd = exec.CommandContext(ctx, "boundary", "authenticate", "-addr", bc.Address, "-format", "json")
+
+	f, err := pty.Start(cmd)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := f.Close()
+		require.NoError(t, err)
+	})
+
+	// Added to prevent inputting login credentials before the authenticate command prompts for
+	// username/password. Without this, the test would print the username and password onto the
+	// terminal as opposed to being hidden by the prompt.
+	time.Sleep(500 * time.Millisecond)
```

Comment:
Is there a better way to handle this?

Reply:
As discussed on Slack, we should try to read from the pty.